packages/grafana-data/src/vector/AsNumberVector.ts

```diff
@@ -9,6 +9,6 @@ import { Vector } from '../types';
 export class AsNumberVector extends Array<number> {
   constructor(field: Vector) {
     super();
-    return field.map((v) => +v) as AsNumberVector;
```

Comment:
@ryantxu not sure about removing these Vector type assertions as i'm not sure exactly what they do right now... thought i'd just do it anyway and ask in this PR 😅